src/impl/pool.jl

```diff
@@ -11,7 +11,11 @@ function max_pooling2d_fwd!(x::AbstractArray{T,4}, y::AbstractArray{T,4},
     hstart = max(hstart, 0) + 1
     wstart = max(wstart, 0) + 1
 
-    y[pw, ph, c, n] = maximum(x[wstart:wend, hstart:hend, c, n])
```

Comment:
Did you try just using a `view`?

Reply:
Look at top comment :p. Anyway, to elaborate, the view gets allocated and the loop is super tight so it has terrible performance.